opencv_ai/splice/io.h

```diff
@@ -0,0 +1,166 @@
+#ifndef OPENCV_AI_SPLICE_IO_H_
+#define OPENCV_AI_SPLICE_IO_H_
+
+#include <opencv_ai/splice/component.h>
+#include <boost/signals2.hpp>
+
+namespace opencv_ai {
+
+template <typename MessageType>
+class ChannelService : public boost::asio::io_service::service {
```

Comment:
Also, do we prefer large umbrella headers over concise headers containing (more or less) single class declarations?